R/DataSetList.R

```diff
@@ -396,48 +396,51 @@ arrange.DataSetList <- function(dsl, ...) {
 }
 
 #' @rdname get_ERT
-#' @param algorithm Which algorithms in the DataSetList to consider.
+#' @param algorithm DEPRECATED, will be removed in next release. Which algorithms in the DataSetList to consider. 
 #' @export
 #'
 get_ERT.DataSetList <-
-  function(ds, ftarget, algorithm = 'all', ...) {
+  function(ds, ftarget, budget = NULL, algorithm = 'all', ...) {
+    if (!missing("algorithm")) warning("Argument 'algorithm' is deprecated and will be removed in the next release of IOHanalyzer.")
     if (algorithm != 'all')
       ds <- subset(ds, algId == algorithm)
-
+    
     rbindlist(lapply(ds, function(ds) {
       res <-
-        cbind(attr(ds, 'DIM'), attr(ds, 'funcId'), get_ERT(ds, ftarget))
+        cbind(attr(ds, 'DIM'), attr(ds, 'funcId'), get_ERT(ds, ftarget, budget))
       colnames(res)[1] <- 'DIM'
       colnames(res)[2] <- 'funcId'
       res
     }))
   }
 
 #' @rdname get_RT_summary
-#' @param algorithm Which algorithms in the DataSetList to consider.
+#' @param algorithm DEPRECATED, will be removed in next release. Which algorithms in the DataSetList to consider. 
 #' @export
 get_RT_summary.DataSetList <-
-  function(ds, ftarget, algorithm = 'all', ...) {
+  function(ds, ftarget, budget = NULL, algorithm = 'all', ...) {
+    if (!missing("algorithm")) warning("Argument 'algorithm' is deprecated and will be removed in the next release of IOHanalyzer.")
```

Comment:
The new logic/usage is to let the user filter the data set list instead of filtering it inside this function?

Reply:
Yes, it was very arbitrary to have filtering on algid in these functions but not any other attributes (and it was never used in any of the other functionality). Having the filtering separate is much clearer in my opinion.

Reply:
Yes, fully agree.